src/app/parser.rs

```diff
@@ -1534,8 +1534,8 @@ impl<'a, 'b> Parser<'a, 'b>
             }
         }
         if let Some(vtor) = arg.validator_os() {
-            if let Err(e) = vtor(val.to_string_lossy().into_owned()) {
-                return Err(Error::value_validation(Some(arg), e, self.color()));
+            if let Err(e) = vtor(val) {
+                return Err(Error::value_validation(Some(arg), e.into_string().unwrap_or("error invalid UTF-8".to_string()), self.color()));
```

Comment:
This will panic if the user returns an `OsString` with invalid UTF-8 (perhaps a path name, and the reason they're using `validator_os` in the first place). So we need to change this to something like `&e.to_string_lossy()`